examples/aes-ccm/src/ccm_mode.rs

```diff
@@ -0,0 +1,197 @@
+use hacspec_lib::*;
+use hacspec_aes::*;
+
+fn format_func(a: &ByteSeq, n: &ByteSeq, p: &ByteSeq, t: u8, alen: u64, nlen: u8, plen: u64) -> ByteSeq {
+    let mut r = 0u64;
+    let mut tmp = 10u64;
+
+    if alen < 0x800000u64 {
+        tmp = 2u64;
+    } else {
+        if alen < 0x100000000u64 {
+            tmp = 6u64;
+        }
+    }
+
+    r = r + ((tmp+alen+15u64)/16u64)+((plen+15u64)/16u64); // ceiling operation used
+    let mut b = ByteSeq::new((16u64*(r+1u64)) as usize);
+
+    // creation of b(0)
+    let qlen: u8 = 15u8 - nlen;
+    let mut flags: u8 = 0u8;
+
+    if alen > 0u64 {
+        flags = 0x40u8;
+    }
+
+    flags = flags | (((t-2u8)/2u8) << 3);
+    flags = flags | (qlen-1u8);
+    b[0] = U8(flags);
+
+    for i in 0..(nlen as usize) {
+        let tmp2 = n.get_exact_chunk(1, i);
+        b = b.set_exact_chunk(1, i+1, &tmp2);
+    }
+
+    let andy: u64 = 255u64; // 0xFF
+    let mut copy: u64 = plen;
+
+    for i in 1..(qlen as usize)+1 {
+        let curr = (copy & andy) as u8;
+        b[16-i] = U8(curr);
+        copy = copy >> 8;
+    }
+
+    // creation of b(1) to b(u)
+    let x: u8 = 0xffu8;
+    let y: u8 = 0xfeu8;
+
+    let mut k = 16u64; // byte number to set next
+    let mut copy2 = alen;
+
+    if alen >= 0x800000u64 {
+        b[16] = U8(x);
+        k = k + 2u64;
+
+        if alen < 0x100000000u64 {
+            b[17] = U8(y);
+        } else {
+            b[17] = U8(x);
+        }
+    }
+
+    for i in 1..(tmp as usize)+1 {
+        let curr = (copy2 & andy) as u8;
+        b[((k+tmp) as usize)-i] = U8(curr);
+        copy2 = copy2 >> 8;
+    }
+
+    k = k + tmp;
+
+    for i in 0..(alen as usize) {
+        let tmp2 = a.get_exact_chunk(1, i);
+        b = b.set_exact_chunk(1, i+(k as usize), &tmp2);
+    }
+
+    k = k + alen;
+
+    for _t in 0..16 {
```

Comment:
There's an issue to fix the typechecker to allow `_` here. Please reference it in a comment here.

Reply:
#135 aims to allow us to replace `_t` with `_` without any objection from the typechecker.